images/virtualization-artifact/pkg/controller/vmsnapshot/vmsnapshot_webhook.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vmsnapshot
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	virtv2 "github.com/deckhouse/virtualization/api/core/v1alpha2"
+)
+
+type Validator struct {
+	logger *slog.Logger
+}
+
+func NewValidator(logger *slog.Logger) *Validator {
+	return &Validator{
+		logger: logger.With("webhook", "validator"),
+	}
+}
+
+func (v *Validator) ValidateCreate(_ context.Context, obj runtime.Object) (admission.Warnings, error) {
+	vmSnapshot, ok := obj.(*virtv2.VirtualMachineSnapshot)
+	if !ok {
+		return nil, fmt.Errorf("expected a VirtualMachineSnapshot but got a %T", obj)
+	}
+
+	if vmSnapshot.Spec.VirtualMachineName == "" {
+		return nil, fmt.Errorf("virtual machine name cannot be empty")
+	}
```

Comment:
I've raised this question before, we decide not to prevent creating resources with  such problems, but to set them into Failed state. Is it OK to prevent creation here?